make-rules/common.mk

```diff
@@ -0,0 +1,198 @@
+COMPONENT_BUILD_ENV += CFLAGS="$(strip $(CFLAGS))"
+endif
+ifneq ($(strip $(CXXFLAGS)),)
+# Only if provided.
+COMPONENT_BUILD_ENV += CXXFLAGS="$(strip $(CXXFLAGS))"
+endif
+# Always process LDFLAGS.
+COMPONENT_BUILD_ENV += LDFLAGS="$(strip $(LDFLAGS))"
+endif
+
+# Assume components built with ant do not need ASLR.
+ifeq ($(strip $(BUILD_STYLE)),ant)
+ASLR_MODE= $(ASLR_NOT_APPLICABLE)
+else
+# Enable ASLR by default unless target build is NO_ARCH.
+ifneq ($(strip $(BUILD_BITS)),NO_ARCH)
+ASLR_MODE= $(ASLR_ENABLE)
+endif # BUILD_BITS NO_ARCH
+endif # BUILD_STYLE ant
+
+# Determine if we should automatically include lint libraries in build and
+# install targets based on presence of llib* files in component directory. This
+# must be done before defining any common targets since the include affects
+# later variable expansion.
+LINT_LIBRARIES ?= $(wildcard llib-*)
+ifneq ($(strip $(LINT_LIBRARIES)),)
+include $(WS_MAKE_RULES)/lint-libraries.mk
+endif
+
+# Default targets; to omit a specific target, explicitly set the related target
+# variable to empty before including this file or set COMMON_TARGETS=no.
+COMMON_TARGETS ?= yes
+
+ifneq ($(strip $(COMMON_TARGETS)),no)
+# For ant-based build style, assume that install is provided by simply building
+# the component and that no tests are available.
+ifeq ($(strip $(BUILD_STYLE)),ant)
+INSTALL_TARGET ?= build
+TEST_TARGET ?= $(NO_TESTS)
+SYSTEM_TEST_TARGET ?= $(NO_TESTS)
+endif
+
+# For archive-based build style, assume that the build, install, and test
+# targets will be provided.
+ifeq ($(strip $(BUILD_STYLE)),archive)
+BUILD_TARGET ?=
+INSTALL_TARGET ?=
+TEST_TARGET ?=
+endif
+
+# For pkg-based build style, assume there are no build, install, or test steps;
+# just a package to be published.
+ifeq ($(strip $(BUILD_STYLE)),pkg)
+BUILD_TARGET=
+INSTALL_TARGET=
+TEST_TARGET=
+SYSTEM_TEST_TARGET=
+build install:
+test system-test:	$(NO_TESTS)
+endif
+
+# If TEST_TARGET is NO_TESTS, assume no system tests by default.
+ifeq ($(strip $(TEST_TARGET)),$(NO_TESTS))
+SYSTEM_TEST_TARGET ?= $(NO_TESTS)
+endif
+
+# If TEST_TARGET is SKIP_TEST, assume system tests are skipped by default.
+ifeq ($(strip $(TEST_TARGET)),$(SKIP_TEST))
+SYSTEM_TEST_TARGET ?= $(SKIP_TEST)
+endif
+
+# Otherwise, attempt to define common targets assuming defaults.
+BUILD_TARGET ?= $(BUILD_$(MK_BITS))
+ifneq ($(strip $(BUILD_TARGET)),)
+build:          $(BUILD_TARGET)
+endif
+
+INSTALL_TARGET ?= $(INSTALL_$(MK_BITS))
+ifneq ($(strip $(INSTALL_TARGET)),)
+install:        $(INSTALL_TARGET)
+endif
+
+TEST_TARGET ?= $(TEST_$(MK_BITS))
+ifneq ($(strip $(TEST_TARGET)),)
+test:           $(TEST_TARGET)
+endif
+
+# For the system-test target, assume that none have been implemented.
+SYSTEM_TEST_TARGET ?= $(SYSTEM_TESTS_NOT_IMPLEMENTED)
+ifneq ($(strip $(SYSTEM_TEST_TARGET)),)
+system-test:    $(SYSTEM_TEST_TARGET)
+endif
+endif # COMMON_TARGETS
+
+# Always needed; every component builds packages.
+include $(WS_MAKE_RULES)/ips.mk
```

Comment:
Do we want to use this Makefile for IPS or other gates? If so, this   this should be conditional.

Reply:
I imagined that gates are a special case and would not rely on this file.
For now introducing this files helps factoring for fonts and allow testing the logic before we extend the use to other components.